firefox-ios/Shared/Extensions/WKWebViewExtensions.swift

```diff
@@ -75,3 +75,14 @@ extension WKUserScript {
         )
     }
 }
+
+
+extension WKFrameInfo {
+    // We check both the top and current frame protocol. This is to also check for mixed content.
+    // For most cases checking the top frame protocol.`hasOnlySecureContent` is deliberatly not used here
+    // since it will return false if any external resource on the page is loaded via
+    // an insecure connection (i.e styles, images, ...), which might be too strict for most applications.
+    public var isFrameLoadedInSecureContext: Bool {
+        return self.webView?.url?.scheme == "https" && self.securityOrigin.protocol == "https"
+    }
```

Comment:
My thought is that we should use this as well for the `LoginsHelper` since we want the same loading context conditions. Is that enough to make it an extension ? Otherwise I am happy to change it and move it to `FormAutofillHelper`. Also you know the code better but `hasOnlySecureContent` is too strict. But anytime we only want to check the protocol for `https` we can use this extension instead.